### DISCUSSION TITLE
v0.22.0

### DISCUSSION THREAD

**Discussion by alex-w** [Stellarium/stellarium#2365] [General]
**The installers are located in the "Assets" section at the end of these release notes!**

[![Github Release](https://img.shields.io/github/downloads/Stellarium/stellarium/v0.22.0/total.svg)](https://github.com/Stellarium/stellarium/releases/v0.22.0)

A huge thanks to the people who helped us a lot by reporting bugs!

## Release notes
The Stellarium team is delighted to announce the release of Stellarium 0.22.0.

The major changes of this version:
- AstroCalc: Eclipse Finder
- AstroCalc: RTS (Rises, Transits, Settings)
- Switch from Bortle scale to physical brightness values for light pollution.
- Allow tweaks for the atmosphere brightness/color model and tone mapping
- Several new calendars in the Calendars plugin
- Seasonal polar caps on Mars
- New and improved sky cultures
- Larger textures allowed in Scenery3D
- Improved OnlineQueries and Satellites plugin
- Many fixes in core and plugins.
- Technical preparations towards a major upgrade of the internal Qt framework.
- There has also been a large number of bug fixes and closed feature requests and enhancements (see full list of changes).

Enjoy!

## What's Changed
Full list of changes:
- Added ability to search satellites by COSPAR ID
- Added ability to support multi supergroups for satellites
- Added checks for valid UTF-8 sequence for SS [ssystem*] files (GH: #2146)
- Added 2 startup shortcuts for linux distros (GH: #2129)
- Added GUIs for tweaking some atmosphere parameters and better understand how these things work together (GH: #2085)
- Added ability switching between Preetham and Schaefer for sky brightness (CIE Y channel) (GH: #2085)
- Added ability switching between 2 presets of chromaticity parameters (Preetham/Stellarium) (GH: #2085)
- Added ability coupling extinction coefficient and atmosphere turbidity (GH: #2085)
- Added an exponential twilight transition to Preetham zenith luminance (GH: #2085)
- Added a flag to suppress atmosphere scattering rendering. Just keeps atmosphere refraction and extinction effects (GH: #2085)
- Added ability tweaking atmosphere: allow solar halo drawing before or after 3d solar sphere (GH: #2085)
- Added ability tweaking atmosphere: allow rendering the Sun after the atmosphere (avoid overbrightening, but may look too dark) (GH: #2085)
- Added GUI to fine-tune viewing environment (tone mapping) (GH: #2085)
- Added scripting functions to modify tone mapping settings (GH: #2085)
- Added color conversion to AdobeRGB (not only sRGB). This *may* give better screen colors on some systems. (GH: #2085)
- Added ability moving StelDialogSeparate to a separate screen (GH: #2132)
- Added human readable format for coordinates of planetary features (GH: #2158)
- Added support decimal coordinates for planetary features (GH: #2158)
- Added Akan (ak), Karelian (krl), Montenegrin (cnr) and Serbian (Latin) (sr@latin) languages support
- Added Lunar Eclipses to Astronomical calculations (GH: #2190, #407)
- Added actions to centering screen on current telescope position (GH: #1956)
- Added proper name for LBN 437
- Added proper name for Stock 2 cluster (GH: #2207)
- Added actions to quick choose of presets for the selected object information (GH: #2206)
- Added Modified Espenak & Meeus (2006, 2022) solution for DeltaT (GH: #2340)
- Added support alternative names (designations) for exoplanets (GH: #2350)
- Added texture for an interesting distorted galaxy NGC 2146 in Camelopardalis
- Added setting of screenshot dpi
- Added New and Astronomical Hindu Calendars (GH: #2284)
- Added Hindu Panchang (GH: #2284)
- Added 60-year-cycle to Tibetan calendar (GH: #2284)
- Added configurable text color (GH: #2284)
- Added mistakenly removed parentheses to the regex (GH: #2324)
- Added Egyptian (Dendera) sky culture (GH: #2329)
- Added Zulu sky culture (GH: #2329)
- Added 2 asterisms: Medium Bull & Minor Bull
- Added Persian Astronomical calendar (GH: #2229)
- Added nowruz to both Persian calendar versions (GH: #2229)
- Added French Revolution Calendar (astronomical version) (GH: #2229)
- Added Tibetan Calendar (GH: #2229)
- Added functions from CC:UE chapter 14 on astronomical functions (GH: #2229)
- Added proper name for CTB 1 (Abell 85) nebula
- Added boundaries to Arabic Lunar Stations skycultute (GH: #2222)
- Added support of seasonal polar caps on the Mars (GH: #2220, #2040)
- Added Revised Julian Calendar into Calendars plugin (GH: #2214)
- Added support heliocentric ecliptic positions of the major planets into AstroCalc/Positions tool (GH: #2233)
- Added Solar Eclipses to Astronomical calculations (GH: #2218, #407)
- Added missing timezone for Nuuk (GH: #2140)
- Added column "Visibility conditions" to the lunar eclipses
- Fixed typos in Satellites plugin
- Fixed Stephenson, Morrison & Hohenkerk solution for DeltaT
- Fixed Stephenson & Houlden (1986) algorithm for DeltaT
- Fixed rendering gravity labels on HiDPI devices
- Fixed rendering gravity labels of compass
- Fixed nomenclature longitudes (GH: #2158)
- Fixed synonyms in DSO names
- Fixed clazy warnings for plugins and core
- Fixed translation the column names in AstroCalc/WUT tool when language is changed
- Fixed jumping to position in search window (GH: #2205)
- Fixed entering geographical coordinates (GH: #2209, #2210)
- Fixed crash on exit when Telescopes tab of Oculars is active (GH: #2103)
- Fixed bug in Meteor showers search dialog (GH: #2356)
- Fixed list of languages
- Fixed typos in sky cultures
- Fixed potential selection bug in AstroCalc/Positions tool
- Fixed visual properties for AstroCalc/Positions tool
- Fixed bug in AstroCalc/Eclipses tool for local solar eclipses when total/annular eclipses take place very close to the horizon (GH: #2345)
- Fixed decimal longitude in AstroCalc/Eclipses tool (GH: #2323)
- Fixed coordinates of few open clusters (DSO Catalog v3.14)
- Fixed crash on start when Calendars plugin not loaded. (GH: #2287)
- Fixed parsing of decimal degrees in AngleSpinBox::stringToDouble() (GH: #2249)
- Fixed light pollution node in TUI (GH: #2253)
- Fixed StelLocationMgr.cpp:parseAngle for negative fractional-degree angles (GH: #2255)
- Fixed reading location from CLI (GH: #2254)
- Fixed planetocentric rectangular coordinates (GH: #2244)
- Fixed R&D solution of DeltaT (GH: #2229)
- Fixed galactic and supergalactic longitudes range in Scripting Engine (GH: #2219)
- Fixed bug in the Hebrew calendar (GH: #2153)
- Fixed errors in Arabic sky cultures
- Fixed computation of rise, transit and set times for the Moon (GH: 2228)
- Fixed alignment for headers in tables
- Fixed security issues: updated jQuery and jQuery UI (GH: #2234, #2235, #2236)
- Fixed bug in AstroCalc/Eclipses where type of local solar eclipses are incorrect (GH: #2237)
- Fixed wrong range of galactic and supergalactic coordinate systems (GH: #2219)
- Changed the languages support: refactoring the code for ISO-639 codes support
- Changed the core: refactoring the code for get angular size of objects (GH: #2180)
- Changed the GUI: compactification of the GUI (GH: #2199)
- Changed AstroCalc/Transits tool: The AstroCalc/Transits tool has been converted to AstroCalc/RTS tool, where RTS is meaning "rises, transits and sets" (GH: #2283)
- Changed regexp in the comets data parser (GH: #2281)
- Changed core: pass Sun, Moon and current planet objects as well as location directly to Atmosphere::computeColor (GH: #2257)
- Changed core: enable multisampling for landscape too (GH: #2260)
- Changed core: switch from Bortle class to luminance as the quantity of light pollution (GH: 2179)
- Updated Arabic sky cultures: renaming sky cultures (GH: #2212)
- Updated Babylonian (MUL.APIN) sky culture
- Updated rules for guessing the groups of satellites
- Updated GUI of Satellites plugin
- Updated default list of TLE sources
- Updated International Designator info for satellites
- Updated default catalog of pulsars
- Updated list of known versions MSVC++
- Updated logger for macOS
- Updated translations
- Updated Stellarium User Guide
- Updated core: replace deprecated QLinkedList by std::list
- Updated core: refactor cardinals (GH: #2352)
- Updated core: refactor qflags (GH: #2351)
- Updated GUI: Improve usability of light pollution display in landscape description (GH: #2296)
- Updated GUI: Prevent negative length of day (GH: #2332)
- Updated labels in Calendars plugin
- Updated defaults for Calendars plugin (GH: #2284)
- Updated Scenery3D plugin: Extend shadowmap/cubemap resolution options 
- Updated list of proper names (GH: #2316, #2319, #2320)
- Updated order of star designations
- Updated shape of asterism "Taurus Poniatovii"
- Updated locations database, nomenclature and common names for DSO (GH: #2262, #2268, #2274, #2280, #2282)
- Updated build instruction
- Updated GUI for AstroCalc/Eclipses tool: allow years before 1582 (GH: #2224)
- Removed default GRS solution (GH: #2187)
- Removed radio frequencies from getInfoMap() method

## Packages notes
package | note
------------|---------
stellarium-0.22.0.tar.gz | Stellarium 0.22.0 for UNIX/Linux (source code). <br />__MD5:__ 43d9c4aea609a02042a0e9a5e1b8fb43 <br />__SHA1:__ c4a00fd756c66fb7df633f496dd3be4300bf1d2b <br />__SHA256:__ 0b4dc23cf9054b5e76cd9bc5ad68e172eb221999e90af37e93667d04fe78c885
stellarium-0.22.0-win32.exe | Stellarium 0.22.0 for Windows (32-bit; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017 Community) with media support. <br /> __MD5__: fd0820cb818e8b639eebad2259934ba5 <br />__SHA1__: 5c26962ec39377b2aa9809cb35f24be043446ad0 <br />__SHA256__: 33deaf2e0d27c977b48c816326482583431eef3d24172012ce52f325334e45b9
stellarium-0.22.0-win64.exe | Stellarium 0.22.0 for Windows (64-bit; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017 Community) with media support. <br /> __MD5__: 85a96c383eb2e38157103d9eedc02176 <br />__SHA1__: f65642b3eefbba35bd7171681b694d656c3e8a66 <br />__SHA256__: 4f29c947940dfb9953f2386c3bfb4b0e91114223c969eba9447187896fe40e9f
Stellarium-0.22.0-x86_64.zip | Stellarium 0.22.0 for Mac OS X (x86_64; Mac OS X 10.12+) based on Qt 5.12.9 (AppleClang 11.0.0) with media support. <br /> __MD5__: 5bd553c5049c8a9a00108ddb584b9455 <br />__SHA1__: 43dd20d3fd792bce62f108fb9ae7b0139311eef2 <br />__SHA256__: 8e55d08d87769c2f02d807d0480839560ad00b93eae74ab2d39a1d5a0541b03d
Stellarium-0.22.0-arm64.zip | Stellarium 0.22.0 for Mac OS X (arm64; Mac OS X 10.12+) based on Qt 5.15.3 (AppleClang 13.1.6) with media support. <br /> __MD5__: 318db1f1ba1dcefec3f7bd25f4001a46 <br />__SHA1__: 793e879d87a9571df746cf7058a4278a62daa1a8 <br />__SHA256__: b9fed25a9272de8300155ad656a527e5878fc1e9fd6a091a14f30830442bbd7d
Stellarium-0.22.0-x86_64.AppImage | Stellarium 0.22.0 for Linux (AMD64; glibc 2.27) based on Qt 5.9.5 (GCC 7.5.0) with media support. <br /> __MD5__: f837852df50961595bce9021210b5874 <br />__SHA1__: a9ec932a7edea6165893306ac2ce0641eafa69cd <br />__SHA256__: af2421bc7391444017847167b6d2384d0bbc18978a254095ae416bd82010478d

<hr /><em>This discussion was created from the release <a href='https://github.com/Stellarium/stellarium/releases/tag/v0.22.0'>v0.22.0</a>.</em>

**Comment by Menno5** [comment]
Sad to see that the Bookmark feature that was suddenly removed and replaced by a (for many useless and/or confusing) Observing Lists thing, has such a low priority and still is not back and keeps on being moved to a next release :( Still don't understand why this step has been done and the easy to work Bookmarks suddenly were gone.
So back to version 21.1 where the bookmarks work just fine.

**Comment by PaulJRead** [comment]
The change to the light pollution implementation is causing havoc for my local users!  The automatic from database setting used to work fine, but now simply gives an orange sky with no objects visible - not the same as previously AND not representative of reality at all, and the manual settings are virtually useless.  All of my group (Leicester Astronomical Society members) are reverting to version 21.3

**Comment by mconsidine** [comment]
I am trying to run this on Linux Mint and getting this error

/tmp/.mount_StellaAYjeKk//usr/bin/stellarium: /tmp/.mount_StellaAYjeKk//usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.12' not found (required by /usr/lib/x86_64-linux-gnu/libQt5Charts.so.5)

The relevant .so file is on the system:
/usr/lib/x86_64-linux-gnu/libQt5Core.so
/usr/lib/x86_64-linux-gnu/libQt5Core.so.5
/usr/lib/x86_64-linux-gnu/libQt5Core.so.5.12
/usr/lib/x86_64-linux-gnu/libQt5Core.so.5.12.8

The prior version of the AppImage started without a problem.

Thoughts?

**Reply by Fellypao** [reply]
Almost the same problem when trying to run the appimage on KUbuntu 21.10:
first:
/tmp/.mount_Stella6A3K4x//usr/bin/stellarium: error while loading shared libraries: libQt5Charts.so.5: cannot open shared object file: No such file or directory

after install libQt5Charts:
/tmp/.mount_Stella69Didg//usr/bin/stellarium: /tmp/.mount_Stella69Didg//usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.15' not found (required by /lib/x86_64-linux-gnu/libQt5Charts.so.5)

**Reply by alex-w** [reply]
@Fellypao @mconsidine please check https://github.com/Stellarium/stellarium/suites/5918466369/artifacts/202175295

**Reply by mconsidine** [reply]
Still seeing this:
./Stellarium-edge-x86_64.AppImage
/tmp/.mount_StelladpGhjp//usr/bin/stellarium: /tmp/.mount_StelladpGhjp//usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.12' not found (required by /usr/lib/x86_64-linux-gnu/libQt5Charts.so.5)

Is there an environment variable or path I should be checking?

Using 'ls' on that location shows:
$ ls /usr/lib/x86_64-linux-gnu/libQt5Charts*
/usr/lib/x86_64-linux-gnu/libQt5Charts.prl
/usr/lib/x86_64-linux-gnu/libQt5Charts.so
/usr/lib/x86_64-linux-gnu/libQt5Charts.so.5
/usr/lib/x86_64-linux-gnu/libQt5Charts.so.5.12
/usr/lib/x86_64-linux-gnu/libQt5Charts.so.5.12.8

Thanks!
mconsidine
PS  when I compiled from source the other day (prior to v0.22), I did not get any errors.

**Reply by gzotti** [reply]
I read this as "libQt5Charts is missing from the AppImage". If you can compile from source (and install the recently added libqt5charts5-dev package before, obviously), it should work.

**Reply by alex-w** [reply]
@Fellypao @mconsidine please check https://github.com/Stellarium/stellarium/suites/5929078993/artifacts/202797190

**Reply by Fellypao** [reply]
@alex-w 
It is working here

**Reply by mconsidine** [reply]
Latest appimage working now.

**Comment by alex-w** [comment]
Hi @PaulJRead @mconsidine @Fellypao! Please check the latest stable version of Stellarium:
https://github.com/Stellarium/stellarium/releases/latest

**Reply by Fellypao** [reply]
Hi @alex-w ,
The latest version runs fine here!

**Comment by PaulJRead** [comment]
Hi Alexander,

Pleased to confirm that the previous issue with v22 seems resolved in 22.1  - Thank you to the team 
 

Regards,

Paul Read


All content of this email including attachments has been scanned for viruses and malware using McAfee Total Protection. The contents are intended only for the addressee, if you are not the intended recipient, please delete this email and any attachments.

 
 

Sent: Sunday, April 17, 2022 at 3:39 PM

**Comment by mconsidine** [comment]
This is working for me now.  Thank you!
mconsidine
